client/src/components/Menu.tsx

```diff
@@ -1,30 +1,47 @@
 import { useEffect, useState } from 'react'
-import { Link, useLocation } from 'react-router-dom'
+import { Link, useLocation, useNavigate } from 'react-router-dom'
+import InputLabel from '@mui/material/InputLabel';
+import MenuItem from '@mui/material/MenuItem';
+import FormControl from '@mui/material/FormControl';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
 
 const linkStyle = { color: '#343434', textDecoration: 'none' }
 const activeLinkStyle = { ...linkStyle, backgroundColor: '#eee', borderRadius: '15px', padding: '10px' }
 const links = [
     { title: 'Home', to: '/', activeSet: new Set(['/']) },
     { title: 'About', to: '/#about', activeSet: new Set(['/#about']) },
     { title: 'Services', to: '/#services', activeSet: new Set(['/#services']) },
+    { title: 'Contact', to: '/#contact', activeSet: new Set(['/#contact']) },
     { title: 'Users', to: '/users', activeSet: new Set(['/users']) },
     { title: 'Groups', to: '/groups', activeSet: new Set(['/groups', '/group']) },
     { title: 'Teams', to: '/teams', activeSet: new Set(['/teams']) },
     { title: 'SharePoint', to: '/sharepoint', activeSet: new Set(['/sharepoint']) },
     { title: 'OneDrive', to: '/onedrive', activeSet: new Set(['/onedrive']) },
     { title: 'Reports', to: '/reports', activeSet: new Set(['/reports']) },
-    { title: 'Contact', to: '/#contact', activeSet: new Set(['/#contact']) },
 ]
 
 export const Menu = () => {
-    const location = useLocation()
-    const [activeLink, setActiveLink] = useState('')
+    const location = useLocation();
+    const [activeLink, setActiveLink] = useState('');
+    const navigate = useNavigate();
+    const [currentPage, setCurrentPage] = useState('');
+    const isLinkAnchor = (linkTo: string) => linkTo === '/' || linkTo.length >= 2 && linkTo.substring(0, 2) === '/#';
```

Comment:
I believe you can simply this logic a bit. 
You can utilize link.startWith("/#") to till if it's a section navigator, or even change the structure of links if needed